Userland/Libraries/LibGUI/DatePickerDialog.h

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2023, David Ganz <david.g.ganz@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/String.h>
+#include <LibCore/DateTime.h>
+#include <LibGUI/Calendar.h>
+#include <LibGUI/Dialog.h>
+#include <LibGUI/Model.h>
+
+namespace GUI {
+
+class DatePickerDialog : public Dialog {
+    C_OBJECT(DatePickerDialog)
+
+public:
+    virtual ~DatePickerDialog() override = default;
+
+    static ExecResult show(Window* parent_window, String title, Core::DateTime& result, Core::DateTime focused_date = Core::DateTime::now());
+
+private:
+    explicit DatePickerDialog(Window* parent_window, String const& title, Core::DateTime focused_date = Core::DateTime::now());
+
+    class MonthListModel final : public GUI::Model {
+    public:
+        enum Column {
+            Month,
+            __Count,
+        };
+
+        static NonnullRefPtr<MonthListModel> create() { return adopt_ref(*new MonthListModel); }
+        virtual ~MonthListModel() override = default;
+
+        virtual int row_count(const GUI::ModelIndex& = GUI::ModelIndex()) const override { return 12; }
+        virtual int column_count(const GUI::ModelIndex& = GUI::ModelIndex()) const override { return Column::__Count; }
+        virtual ErrorOr<String> column_name(int) const override;
+        virtual GUI::Variant data(const GUI::ModelIndex&, GUI::ModelRole) const override;
```

Comment:
Nit-pick: Please use east const: `Foo const&` not `const Foo&`.

(clang-format is not great at this, and the IDE I use also messes this up constantly, so it's easily done.)